lms/djangoapps/mobile_api/utils.py

```diff
@@ -30,12 +34,28 @@ def _wrapper(self, request, *args, **kwargs):
             """
             course_id = CourseKey.from_string(kwargs.pop('course_id'))
             with modulestore().bulk_operations(course_id):
-                course = get_course_with_access(
-                    request.user,
-                    'load_mobile' if verify_enrolled else 'load_mobile_no_enrollment_check',
-                    course_id,
-                    depth=depth
-                )
+                try:
+                    course = get_course_with_access(
+                        request.user,
+                        'load_mobile' if verify_enrolled else 'load_mobile_no_enrollment_check',
+                        course_id,
+                        depth=depth
+                    )
+                except Http404 as error:
+                    # any_unfulfilled_milestones called a second time since get_course_with_access returns a bool
+                    if any_unfulfilled_milestones(course_id, request.user.id):
```

Comment:
Side comment: I really don't like us having specialized code here to handle milestones.  However, until the `has_access` method is improved to provide additional context and not just return a boolean, I don't see another way.  :(